Bloom.md

```diff
@@ -0,0 +1,341 @@
+false positive rate that the user configures.
+
+When a user adds an item (e.g. BF.ADD) to the filter, the item is passed through the filter and corresponding bit are set
+to 1. When a user checks whether an item exists on a filter (e.g. BF.EXISTS), the item is passed through the filters
+and if all the resolved bits have values as 1, we can say that the item exists with a false positive rate of
+X (specified by the user when creating the filter). If any of the bits are 0, the item does not exist and the BF.EXISTS
+operation will return 0.
+
+We have the following terminologies / properties:
+* Bloom Object: The top level structure representing the data type. It contains meta data and a list of lower-level
+                Bloom Filters (Implemented by an external crate) in case of scaling and a single lower-level bloom filter
+                in case of non scaling.
+* Bloom Filter: A single bloom filter (Implemented by an external crate).
+* Capacity: The number of items we expect to add to a bloom filter. This controls the size of the filter.
+* False Positive Rate: The accuracy the user expects when operating (set / check) on a filter. This controls the number
+                        of hash functions.
+* Expansion Rate: This is used in scalable bloom filters where multiple bloom filters are stacked to allow users to
+                    continue using the same bloom object when it reaches capacity by adding another filter of larger
+                    capacity (expansion rate * prev filter capacity).
+* Hash Functions: Hash functions used for bit check and set operations underneath.
+* Hash Key: Keys used by the hashing function.
+
+
+### Module OnLoad
+
+Upon loading, the module registers a new bloom module based data type, creates bloom filter (BF.*) commands,
+bloom specific configurations and the bloom ACL category.
+
+* Module name: bloom
+* Data type name: bloom
+
+### Persistence
+
+ValkeyBloom implements persistence related Module data type callbacks for the Bloom data type:
+
+* rdb_save: Serializes bloom objects to RDB.
+* rdb_load: Deserializes bloom objects from RDB.
+* aof_rewrite: Emits commands into the AOF during the AOF rewriting process.
+
+### RDB Compatibility with ReBloom:
+
+During RDB Save of a bloom object, we will need to save the number of filters, expansion rate, false positive rate.
+And for every underlying bloom filter in this object, we store its hash keys used by hashing functions, number of
+hashing functions, number of bits of the bit array, bytes of the bit array itself.
+
+The data above is written into the RDB during save operations and can be restored from RDB load. However, the data that
+gets written to the RDB is specific to the data type's structure and struct members. Additionally, the data within the
+underlying bloom filter (from the external crate) is specific to the implementation of the bloom filter as the
+hash key (seed), hashing algorithm, raw bit array data, etc. can all vary.
```

Comment:
Did we decide if we are randomly generating a seed or not?

Reply:
I updated the section to explain that we will use a fixed seed